cmd/frontend/internal/app/cluster_debug.go

```diff
@@ -0,0 +1,224 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ericchiang/k8s"
+	corev1 "github.com/ericchiang/k8s/apis/core/v1"
+	"github.com/sourcegraph/sourcegraph/internal/env"
+	"github.com/sourcegraph/sourcegraph/internal/errcode"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+type proxyEndpoint struct {
+	reverseProxy http.Handler
+	service      string
+	displayName  string
+	host         string
+}
+
+func (pe *proxyEndpoint) String() string {
+	var sb strings.Builder
+
+	sb.WriteString(pe.service)
+	sb.WriteString("-")
+	sb.WriteString(pe.host)
+
+	return sb.String()
```

Comment:
Feel like an overkill with a `strings.Builder` :D

```suggestion
	return fmt.Sprintf("%s-%s", pe.service, pe.host)
```